components/x-teaser/src/concerns/image-service.js

```diff
@@ -1,14 +1,19 @@
 const BASE_URL = 'https://www.ft.com/__origami/service/image/v2/images/raw';
-const OPTIONS = ['source=next', 'fit=scale-down', 'dpr=2'];
+const OPTIONS = Object.freeze({
+	source: 'next', 
+	fit: 'scale-down',
+	dpr:2,
+});
 
 /**
  * Image Service
  * @param {String} url
  * @param {Number} width
  * @param {String} options
  */
-export default function imageService(url, width, options) {
+export default function imageService(url, width, addOptions, overrideOptions={}) {
```

Comment:
I don't really get why two different but almost identical params are needed? Perhaps you could refactor the other call sites to pass in an object rather than a string for this argument?

I like the use of the URLSearchParams class, although you should get an instance of one when you create an instance of a URL. so you could do something like this...

```js
const result = new URL(`${BASE_URL}/${encodeURIComponent(url)}`);
if (typeof options === 'string') { // might not be needed if this is refactored?
 options = Object.fromEntries(options.split('&').map(s => s.split('=')))
}
result.search = new URLSearchParams({...OPTIONS, ...options })
result.searchParams.set('width', width);
return result.href;
```
This ensures that a query param isn't set twice (which I don't think is valid for the image service).

Reply:
this is nice code 😎